drivers/pwm/pwm_nxp_s32_emios.c

```diff
@@ -0,0 +1,445 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <Emios_Mcl_Ip.h>
+#include <Emios_Pwm_Ip.h>
+
+#define LOG_MODULE_NAME nxp_s32_emios
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_PWM_LOG_LEVEL);
+
+#define DT_DRV_COMPAT	nxp_s32_emios_pwm
+
+/* Macros used to glue devicetree with RTD's definition */
+#define BUS_A				EMIOS_PWM_IP_BUS_A
+#define BUS_B				EMIOS_PWM_IP_BUS_BCDE
+#define BUS_C				EMIOS_PWM_IP_BUS_BCDE
+#define BUS_D				EMIOS_PWM_IP_BUS_BCDE
+#define BUS_E				EMIOS_PWM_IP_BUS_BCDE
+#define BUS_F				EMIOS_PWM_IP_BUS_F
+
+#define EMIOS_MASTER_BUS_MODE(mode)	DT_CAT(EMIOS_IP_, mode)
+#define EMIOS_PWM_MODE(mode)		DT_CAT(EMIOS_PWM_IP_, mode)
+#define EMIOS_PWM_MODE_WITH_FLAG(mode)	DT_CAT3(EMIOS_PWM_IP_, mode, _FLAG)
+#define EMIOS_PWM_POLARITY(mode)	DT_CAT(EMIOS_PWM_IP_, mode)
+#define EMIOS_PWM_PRESCALER_SRC(mode)	DT_CAT(EMIOS_PWM_IP_PS_SRC_, mode)
```

Comment:
Nit- can we move these macros to the bottom of the file, where they are used?

Reply:
I cleaned up a little bit and move to bottom of the file